tests/round_labeling/test_all_paths_comprehensive.py

```
@@ -0,0 +1,439 @@
+class PathGenerator:
+    """Efficient path generator with caching and filtering."""
+
+    def __init__(self, constraints: Optional[PathConstraints] = None):
+        self.constraints = constraints or PathConstraints(
+            min_length=CONFIG.min_rounds + 2,  # +2 for START and END
+            max_length=CONFIG.max_rounds + 2,
+            source_node="START",
+            target_node="END",
+        )
+
+    @lru_cache(maxsize=1)
```

Ruff (0.11.9)

Use of `functools.lru_cache` or `functools.cache` on methods can lead to memory leaks

(B019)